### ISSUE TITLE
Incorrect check for partition in probe_all()

### ISSUE THREAD

**Issue by arvidjaar** [karelzak/util-linux#728]
Quoting https://github.com/karelzak/util-linux/blob/6bf071afa2ca3d7853aa76a702415fa77c1719e5/libblkid/src/devname.c#L508

> ends in a digit, clearly a partition

That is obviously wrong for NVMe devices which look like `/dev/nvme0n1`. For such devices kernel adds suffix `pN` to build partition name: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/block/partition-generic.c#n35

I hit this when adding support for NVMe to BootInfoScript. Code trusts `blkid` to not display full device when partitions exist in cache; for NVMe `blkid` displays both full device and partitions, which is different from other block devices.

**Comment by karelzak** [comment]
Good point. It would be better to use /sys based whole-disk detection (sysfs_devno_is_wholedisk() -- as we already use on another places), rather than depend on device name format.

This blkid code is pretty old, older than /sys ;-)
